### ISSUE TITLE
Token not refreshed in entra id (azure provider)

### ISSUE THREAD

**Issue by thomas-jaeger** [verbb/social-login#30]
### Describe the bug

Login fails after session times out:

'social-login:error' => 'Unable to process callback for “azure”: “interaction_required
AADSTS70043: The refresh token has expired or is invalid due to sign-in frequency checks by conditional access. The token was issued on 2024-05-26T07:18:14.5164744Z and the maximum allowed lifetime for this request is 43200."

### Steps to reproduce

1. Log in with Entra ID/Azure with conditional access enabled
2. Wait until the session times out
3. Click the Social Login button
4. The browser is redirected to Entra ID and then directly back to Craft
5. The mentioned error above is printed to the console, and the user remains on the login screen

### Craft CMS version

4.9.5

### Plugin version

1.0.14

### Multi-site?

Yes

**Comment by engram-design** [comment]
From what I've read, this is caused by the use of conditional access policies on the Azure account. Further reading [here](https://learn.microsoft.com/en-us/azure/active-directory/conditional-access/howto-conditional-access-session-lifetime#user-sign-in-frequency), so it might be worth checking with that?

**Comment by thomas-jaeger** [comment]
I have tracked down this issue a little bit and after comparing with an other working product, the main difference i spotted is the parameter approval_prompt. According to the microsoft docs, there is a similiar parameter named prompt (https://learn.microsoft.com/de-de/entra/identity-platform/v2-oauth2-auth-code-flow) and no parameter approval_prompt.
The other product i checked used
prompt=select_account
instead of
approval_prompt=auto

**Comment by engram-design** [comment]
Interesting, thanks for the investigation. I wonder if this is an Azure to Entra change. I’ll look into it!

**Comment by thomas-jaeger** [comment]
I added the following code to vendor/verbb/auth/src/clients/azure/provider/Azure.php

```
    protected function getAuthorizationParameters(array $options)
    {
        $options = parent::getAuthorizationParameters($options);
        $options['prompt'] = 'select_account';
        unset($options['approval_prompt']);
        return $options;
    }
```

Might be a little bit hacky, but now it works.

**Comment by engram-design** [comment]
Thanks for investigating this! Added for the next release. To get this early, run `composer require verbb/social-login:"dev-craft-4 as 1.0.15" -W`.

**Comment by engram-design** [comment]
Fixed in [1.0.16](https://github.com/verbb/social-login/releases/tag/1.0.16)
